src/sentry/eventstream/kafka/backend.py

```diff
@@ -239,7 +229,42 @@ def _build_consumer(
         )
         return consumer
 
-    def run_batched_consumer(
+    def _build_streaming_consumer(
+        self,
+        consumer_group: str,
+        topic: str,
+        commit_log_topic: str,
+        synchronize_commit_group: str,
+        commit_batch_size: int,
+        commit_batch_timeout_ms: int,
+        concurrency: int,
+        initial_offset_reset: Union[Literal["latest"], Literal["earliest"]],
+        strict_offset_reset: Optional[bool],
+    ) -> StreamProcessor[KafkaPayload]:
+
+        cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+
+        configuration = build_kafka_consumer_configuration(
+            get_kafka_consumer_cluster_options(cluster_name),
+            group_id=consumer_group,
+            auto_offset_reset=initial_offset_reset,
+            strict_offset_reset=strict_offset_reset,
+        )
+
+        synchronized_consumer = ArroyoSynchronizedConsumer(
+            consumer=KafkaConsumer(configuration),
+            commit_log_consumer=KafkaConsumer(configuration),
+            commit_log_topic=Topic(commit_log_topic),
+            commit_log_groups={synchronize_commit_group},
+        )
+
+        strategy_factory = PostProcessForwarderStrategyFactory(concurrency, 1000)
+
+        return StreamProcessor(
+            synchronized_consumer, Topic(topic), strategy_factory, ONCE_PER_SECOND
+        )
+
+    def run_consumer(
         self,
         entity: Union[Literal["all"], Literal["errors"], Literal["transactions"]],
```

Comment:
IIRC, we cannot run both the entities together anymore. If that is the case, we can remove the all option

Reply:
Done in https://github.com/getsentry/sentry/pull/40206